libs/platform/src/lib/components/approval-flow/approval-flow.component.ts

```diff
@@ -290,61 +550,159 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
         graph[0] = { nodes: [rootNode] };
         let index = 1;
         let foundLastStep = false;
+        const _metaMap: { [key: string]: ApprovalGraphNodeMetadata } = {};
+
+        // meta 1st run
+        nodes.forEach(n => {
```

Comment:
This method is a monster, I lost track :)

Reply:
yea it's huge, I agree. I split it in two and moved most of metadata calculations to a separate method